src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/WeakEvent`1.cs

```diff
@@ -3,129 +3,57 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Collections.Immutable;
-using System.Runtime.CompilerServices;
 
 namespace Roslyn.Utilities;
 
 internal sealed class WeakEvent<TEventArgs>
 {
+    private readonly object _guard = new();
```

Comment:
could be removed with the SyncLock idea from above.

Reply:
Not sure what you mean.